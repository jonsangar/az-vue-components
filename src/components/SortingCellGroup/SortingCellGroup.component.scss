.az-sorting-group{
  @include shadow(2);

  overflow: hidden;
  border-radius: $border-radius;
}
