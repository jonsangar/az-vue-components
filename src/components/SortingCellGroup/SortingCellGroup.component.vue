<template lang="pug">
.az-sorting-group
  SortingCell(
    v-if="title"
    :itemID="itemID"
    :name="title"
    :settingsActive="settingsActive"
    :activeHandle="activeHandle"
    :validation="validation"
    :isHeader="true"
    :keyboardAdjust="keyboardAdjust"
    @focus="activeEditMode"
    @save="saveData")
  slot
</template>

<script>
import SortingCell from '../SortingCell';

export default {
  components: { SortingCell },

  props: {
    title: {
      type: String,
      default: ''
    },

    validation: {
      type: String,
      default: ''
    },

    settingsActive: {
      type: Boolean,
      default: false
    },

    activeHandle: {
      type: Boolean,
      default: true
    },

    itemID: {
      type: String,
      required: false
    },

    keyboardAdjust: {
      type: Boolean,
      default: false
    }
  },

  methods: {
    saveData(data) {
      this.$emit('save',data);
    },

    activeEditMode() {
      this.$emit('focus');
    }
  }

}
</script>

<style lang="scss" src="./SortingCellGroup.component.scss" scoped></style>
