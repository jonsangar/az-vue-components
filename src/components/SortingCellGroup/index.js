import SortingCellGroup from './SortingCellGroup.component.vue';

export default SortingCellGroup;
