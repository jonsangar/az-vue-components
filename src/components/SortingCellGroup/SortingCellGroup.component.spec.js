import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import SortingCellGroup from './SortingCellGroup.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('SortingCellGroup.vue', () => {
  const build = () => {
    const wrapper = shallowMount(SortingCellGroup, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente SortingCellGroup con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que emite el evento focus cuando se activa el modo edición', async () => {
    const { wrapper } = build()

    await wrapper.vm.activeEditMode()
    expect(wrapper.emitted().focus).toBeTruthy()
    expect(wrapper.emitted().focus.length).toBe(1)
    expect(wrapper.emitted().focus[0]).toEqual([])
  });

  it('Comprueba que emite el evento save cuando se ejecuta saveData', async () => {
    const { wrapper } = build()

    const data = {name: 'Rafa'}
    await wrapper.vm.saveData(data)
    expect(wrapper.emitted().save).toBeTruthy()
    expect(wrapper.emitted().save.length).toBe(1)
    expect(wrapper.emitted().save[0]).toEqual([data])
  });
});
