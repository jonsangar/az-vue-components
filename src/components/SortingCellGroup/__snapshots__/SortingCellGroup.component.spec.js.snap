// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingCellGroup.vue Comprueba renderizado del componente SortingCellGroup con snapshot 1`] = `
<div class="az-sorting-group">
  <!---->
</div>
`;
